src/me/mrCookieSlime/Slimefun/Objects/MultiBlock.java

```diff
@@ -12,18 +13,56 @@
 public class MultiBlock {
 	
 	private Material[] blocks;
-	private Material trigger;
-	
+	private BlockFace trigger;
+	private boolean isSymmetric;
+
+	@Deprecated
 	public MultiBlock(Material[] build, Material trigger) {
+		this.blocks = build;
+		this.isSymmetric = isSymmetric(build);
+		//Hacky, the other constructor should be used
+		for (int i = 0; i < 9; i++)
+		{
+			if (trigger.equals(build[i]))
+			{
+				switch (i) {
+					case 1:
+						this.trigger = BlockFace.DOWN;
+						break;
+					case 4:
+						this.trigger = BlockFace.SELF;
+						break;
+					case 7:
+						this.trigger = BlockFace.UP;
+						break;
+				}
+				break;
+			}
+		}
+	}
+	
+	public MultiBlock(Material[] build, BlockFace trigger)
+	{
 		this.blocks = build;
 		this.trigger = trigger;
+		this.isSymmetric = isSymmetric(build);
+	}
+	
+	private boolean isSymmetric(Material[] blocks)
+	{
+		if (blocks[0] == blocks[2]
+			&& blocks[3] == blocks[5]
+			&& blocks[6] == blocks[8]) {
+			return true;
+		}
+		return false;
```

Comment:
You can replace this if-statement by just returning the boolean-expression inside the brackets.